### ISSUE TITLE
Showcase Section - Layout, Technical Implementation, Workflow

### ISSUE THREAD

**Issue by Merelleya** [HaxeFoundation/haxe.org#101]
Let's move the general discussion about how we want the showcase section to look and where it should be displayed to this issue and leave #98 for the Article about Dan Hett.

Quick Summary

* We agree to have a Showcase
* We are discussing layout
* We will aim for one showcase a month

The layout:
* should support pictures and videos
* should support highlighted quotes
* descriptions or comment text below photos and videos
* be one column
* should also be readable on mobile devices (responsive)
* maybe have several highlighting and position options for text/quotes
* support headlines and sub headlines as well as commentary text (italics)

I will think about the editing/writing workflow and make some notes on that. We can have one issue for idea collection. Once an idea is picked up, we might want to have a separate issue for that idea so that we keep it neat and get to complete it.

**Comment by ncannasse** [comment]
I'm fine with using a third party blog system for these, and link them from haxe.org.
For instance i find medium.com to be a nice way to read such formats

**Comment by Simn** [comment]
I would prefer if we use a Haxe-based solution instead. I can train @Merelleya to use some markdown which should really be enough to get nicely formatted one-column articles.

**Comment by markknol** [comment]
You could look on how haxe.io loads its content, thats also a blog that uses markdown.

**Comment by skial** [comment]
> You could look on how haxe.io loads its content, thats also a blog that uses markdown.

I wouldn't, its still broken at the moment. _If_ you want to know how it used to work you can [read](https://github.com/skial/haxe.io/issues/97) the long explanation over in the haxe.io issues.

:+1: for haxe.org news content being markdown based.

**Comment by Simn** [comment]
Let's not overthink this, we already support displaying markdown. In fact, the entire manual displaying is based on markdown.

All we need is a view without side-bar and a few convenience CSS classes for floating text around an image with some sensible margin.

@jasononeil: I remember that you said something about the markdown library not supporting something, but I don't recall any details. Is there anything we should report upstream?

**Comment by jasononeil** [comment]
We could do Markdown easily enough, especially if @Merelleya is the main content editor and @Simn is able to help as needed.  

I'm not sure which limitation I mentioned.  One that comes to mind is an issue with Markdown itself, rather than haxe-markdown: It has no built-in function to give a paragraph a specific CSS class, like `<blockquote class='pullout-quote'>` or similar.  You could just slot some HTML in, or we come up with a rule, like "every blockquote is displayed as a pullout quote, and we choose where it will be located".

We'd have to figure out the strategy for photos and videos too.  If we develop our own solution we'll probably have to crop/resize images manually, unless there's a good open source tool someone wants to recommend.  To be fair though, manually editing them is fairly fast and usually gives a much higher quality result.

All the other points on Josefiene's list should be achievable with a basic markdown parser though.

For the record I'm not opposed to using an external service like Medium, but I also think if we have the feature-set carefully defined we could implement this without much code.

**Comment by Simn** [comment]
We want to promote Haxe for web which isn't off to a good start if our own page doesn't use it for something relatively simple such as this.

You are right about the cropping/resizing, but I agree it can be done offline easily enough. I'm more concerned about getting some preview going locally.

**Comment by Merelleya** [comment]
Yeah. I will definitely need a preview function. There will be hell otherwise. Ideally the preview would even be shareable....

**Comment by Simn** [comment]
print screen + upload is very shareable...

**Comment by markknol** [comment]
The (=this) comment-box of Github supports drag-drop of images very nicely, it one could build such that would be awesome. With a preview.

**Comment by Simn** [comment]
Another nice thing to have would be translations. At least for the languages where English tends to be a problem: Spanish, French, Russian, Japanese. If we don't make these articles too wordy I'm sure we can find people in the community to help with that and then have some flags at the top of the articles.

**Comment by Merelleya** [comment]
I am aiming for 800 to 1000 words. Anything else is very long to read on a a web page (I dealt with a 2500 word article the other day and found it extremely long). I can translate to Spanish myself. It won't be perfect but good enough, I hope. I am very much for translation into Russian since we have a big Russian community we never hear from due to language barriers. Russian speakers are actually our third biggest audience after English and  French (We have about 20k English speakers and then 1,9k French and about 1,8k Russian and German).

**Comment by Merelleya** [comment]
If we look at location and not language, Russia is on the second place directly after the US.

**Comment by Simn** [comment]
Maybe @nadako could help with the Russian translations?

**Comment by jasononeil** [comment]
> We want to promote Haxe for web which isn't off to a good start if our own page doesn't use it for something relatively simple such as this.

Fair point, and we certainly can achieve this kind of thing in Haxe, it's more a question of spending time on it vs using an off the shelf solution.  After I've finished working on the haxelib site I'll do a quick test to see if we can have a relatively elegant writing process in front-end Haxe/JS with integrated image uploads etc.  Shouldn't be too much work, but I can spend an afternoon finding out if it will be good enough for our use case.

One other alternative worth mentioning is that we could use [ckeditor](http://ckeditor.com/) (with some [externs](https://github.com/tjrhodes/ckeditor-externs).  They have some drag-drop upload plugins, one of which my co-worker is working on integrating with ufront uploads (hopefully this week).

Personally I prefer writing in Markdown, but it's worth putting the option out there.

**Comment by ncannasse** [comment]
I think that's quite a waste of time to implement our own blog system, with correct editing, all the media features correctly tuned (facebook, twitter, keywords generation etc.) and mobile steaming of pictures (needs resizing etc.), I'm quite sure we would not get that right in the first place, so why not using something that works well and is maintained by someone else ?

Also, I think hosting on somewhere else than haxe.org makes it actually "feels" better, more like a 3rd party article and not marketing material produced by the website owner. We have more chances it will be shared this way...

**Comment by Simn** [comment]
Isn't that just deception? "We" write these articles so there's no point in trying to hide that fact. In general I like to be very honest about Haxe and I have no problem pointing out what I consider its weaknesses. I also believe a "Things we could improve upon" article would be received very positively.

As for the blog system: I agree for the moment. I really just want to have some markdown files that are nicely represented on the website. We can look into making our own wordpress for ufront a bit later. ;)

**Comment by ncannasse** [comment]
@Simn all marketing can be seen as some kind of deception, since you're purposely trying to put forward your strong points. I have no problem pointing weaknesses as well, but only when people are already quite interested in it. You don't start by presenting your weaknesses first for instance

**Comment by Merelleya** [comment]
Here are some notes on my editing/writing workflow. I am currently prepping the TiVo Showcase and an Article about Jonas Nyström and his sheet-music project.


https://docs.google.com/document/d/1L34Npalgm3SIKdXiSCISWQLVkLhjmNUr-QwtcevVgoY/edit?usp=sharing

Generally, prepping takes the longest, depending on how much content there is to review. Also, if communication is slow, it may take longer than a month to pull one article together, which is why I like to be working on more than one.

If you have any more ideas, just let me know.

**Comment by Simn** [comment]
Showcase articles are in the making now so we can close this issue.
